conda_build_prepare/prepare.py

```diff
@@ -136,10 +137,42 @@ def _try_to_get_git_output(cmd_string):
         YAML().dump(metadata, meta)
 
 
+def _get_latest_mtime_in_dir(directory, time_format):
+    all_dir_mtimes = []
+    for _dir,_,files in os.walk(directory):
+        for f in files:
+            file_mtime = os.path.getmtime(os.path.join(_dir, f))
+            all_dir_mtimes.append(file_mtime)
+    latest_timestamp = max(all_dir_mtimes)
+    latest_datetime_utc = datetime.fromtimestamp(latest_timestamp, timezone.utc)
+    return latest_datetime_utc.strftime(time_format)
+
+
+def _set_date_env_vars(recipe_dir):
+    time_formats = {
+        'DATE_NUM': '%Y%m%d%H%M%S',
+        'DATE_STR': '%Y%m%d_%H%M%S',
+    }
+    absent_vars=[v for v in ['DATE_NUM', 'DATE_STR'] if v not in os.environ]
+
+    for absent_var in absent_vars:
```

Comment:
this is actually quite suboptimal

- we have two vars, that are potentially inconsistent. Is this ok?
- we walk the tree twice, if we have no vars and no git repo - but we get the same result, only in a different format

Reply:
You're right that it asking twice can be enhanced. I've changed it so now git or filesystem is only asked once.

About the inconsistence: I wanted to make it possible to set only `DATE_STR` and not have it overwritten because there's no `DATE_NUM`.

I'm not so sure that these variables really need to be in sync but I guess it doesn't hurt so now `DATE_NUM` always consists of all digits from `DATE_STR` (either exported before running *cbp* or set by *cbp*): https://github.com/litex-hub/conda-build-prepare/pull/3/files#diff-af79d43d7d1d330c6de2011b62c382b72c68d7721fa429a2366d7e5e2e3cf7d5R164

I've just checked that even if `DATE_STR` has no digits, then resulting empty `DATE_NUM` is still acceptable by `conda-build`.